packages/tupaia-web/src/features/Dashboard/DashboardMenu.tsx

```diff
@@ -73,15 +74,78 @@ const DashboardMenuItem = ({ dashboardName, onClose }: DashboardMenuItemProps) =
   );
 };
 
+interface ActionsMenuProps {
+  setExportModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const ActionsMenu = ({ setExportModalOpen}: ActionsMenuProps) => {
+  const [isSubscribed, setIsSubscribed] = useState<boolean>(false)
+  const [mailingListEnabled, setMailingListEnabled] = useState<boolean>(false)
+  const { projectCode, entityCode, dashboardName } = useParams();
+  const {
+    activeDashboard,
+  } = useDashboards(projectCode, entityCode, dashboardName);
+  
+
+
+  const handleSubscribe = (isUserSubscribed) => {
+    // TODO: Add subscribe and unsubscribe sequence
+    // Add react query to POST request for add or remove subscription
+    // Optional - add loading indicator to option.
+    console.log(isUserSubscribed)
+  } 
+  useEffect(() => {
+    const { mailingLists } = activeDashboard;
+    const { mailingListEntity, userSubscribed } = mailingLists.find(({mailingListEntityCode}) => {
+    mailingListEntityCode === entityCode
+    })
+    if(!mailingListEntity) {
+      setMailingListEnabled(false)
+      setIsSubscribed(false)
+    }
+    if(!userSubscribed) {
+      setMailingListEnabled(true)
+      setIsSubscribed(false)
+    }
+    setIsSubscribed(true)
+  },[activeDashboard])
+
+  const exportOption: ActionsMenuOptionType = { label: 'Export', action: () => setExportModalOpen(true), ActionIcon: () => <ExportIcon fill="white"/>, toolTipTitle: 'Export dashboard' }
+  const menuOptions: ActionsMenuOptionType[] = [exportOption]
+  
+  if(!mailingListEnabled) {
+    return (
+      <UIActionsMenu options={menuOptions} includesIcons={true} />
+    )
+  }
+
+  const subscribeOption = {
+    label: isSubscribed ? 'Subscribed' : 'Subscribe', 
+    action: () => handleSubscribe(isSubscribed), 
+    ActionIcon: isSubscribed ? CheckCircleIcon : AddCircleOutlineIcon, 
+    toolTipTitle: isSubscribed ? 'Remove yourself from email updates' : 'Subscribe to receive dashboard email updates', 
+    color: isSubscribed ? 'primary' : undefined,
+  }
+
+  menuOptions.push(subscribeOption)
+
+  return (
+    <UIActionsMenu options={menuOptions} includesIcons={true} />
+  )
```

Comment:
I find it just a little cleaner to add the subscribe button to the options if a mailing list is present
```suggestion
  if (mailingList) {
    const { isSubscribed } = mailingList;
    const subscribeOption = {
      label: isSubscribed ? 'Subscribed' : 'Subscribe',
      action: () => handleSubscribe(isSubscribed),
      ActionIcon: isSubscribed ? CheckCircleIcon : AddCircleOutlineIcon,
      toolTipTitle: isSubscribed
        ? 'Remove yourself from email updates'
        : 'Subscribe to receive dashboard email updates',
      color: isSubscribed ? 'primary' : undefined,
    };

    menuOptions.push(subscribeOption);
  }

  return <UIActionsMenu options={menuOptions} includesIcons={true} />;
```

Reply:
Yep, all makes sense, thanks @rohan-bes . Have pushed up some commits.